### PR TITLE
add Switch docs to site

### LINKED ISSUE #2289: Switch Documentation

### PR DESCRIPTION
**Pull request by feliciackh**
Closes #2289

### CODE REVIEW

File: site/docs/components/switch/accessibility.mdx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Leave the frontmatter as is
+title:
+  $ref: ./#/title
+layout: DetailComponent
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
```
Comment (lilyvc) [inline site/docs/components/switch/accessibility.mdx:11]: heading?

File: site/docs/components/switch/accessibility.mdx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+- On focus, the screen reader should:
+  - Identify the switch
+  - Read out the text
+  - Express the state
+
```
Comment (lilyvc) [inline site/docs/components/switch/accessibility.mdx:16]:
```suggestion

### Keyboard interactions

```

File: site/docs/components/switch/accessibility.mdx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  $ref: ./#/title
+layout: DetailComponent
+sidebar:
+  exclude: true
+data:
+  $ref: ./#/data
+---
+
+- On focus, the screen reader should:
+  - Identify the switch
```
Comment (lilyvc) [inline site/docs/components/switch/accessibility.mdx:13]:
```suggestion
  - Identify the Switch
```

Comment (feliciackh) [reply site/docs/components/switch/accessibility.mdx:13]: keeping it lowercase for now as per content team's guidance

File: site/docs/components/switch/accessibility.mdx (hunk 1 above)
Comment (feliciackh) [inline site/docs/components/switch/accessibility.mdx:11]:
```suggestion

### Best practices
```

**Review by Fercas123 — DISMISSED** [review]

File: site/docs/components/switch/examples.mdx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+- Only use the `checked` prop for a controlled switch that requires some logic for validating value.
+- We recommend that you always use a text description for a switch.
+
+</LivePreview>
+
+<LivePreview componentName="switch" exampleName="DefaultChecked" displayName="Checked" >
+
+### Checked
+
+When ` defaultChecked=”true”`, switch is in a `checked` state and the switch thumb sits to the right side of the track.
```
Comment (JoanaMMoreira) [inline site/docs/components/switch/examples.mdx:33]:
```suggestion
When `defaultChecked=”true”`, switch is in a `checked` state and the switch thumb sits to the right side of the track.
```

File: site/docs/components/switch/usage.mdx [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+### Props
+
+{/* Update packageName and componentName below as needed */}
+{/* packageName is optional and defaults to "core" if omitted */}
+
+<PropsTable packageName="lab" componentName="Switch" />
+
+### References
+
+- Switch Pattern (https://www.w3.org/WAI/ARIA/apg/patterns/switch/ ) W3C
```
Comment (JoanaMMoreira) [inline site/docs/components/switch/usage.mdx:51]:
```suggestion
- [Switch Pattern W3C](https://www.w3.org/WAI/ARIA/apg/patterns/switch/)
```

Comment (feliciackh) [reply site/docs/components/switch/usage.mdx:51]: i think design wanted it displayed this way as is the case for a few other components, will leave it as is for now

**Review by JoanaMMoreira — DISMISSED** [review]

File: site/docs/components/switch/accessibility.mdx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  - Read out the text
+  - Express the state
+
+### Keyboard interactions
+
+<KeyboardControls>
+<KeyboardControl keyOrCombos="Tab">
+
+- If focus is on the previous element in the tab order, move focus onto Switch.
+- If focus is on Switch, moves focus to the next focusable element within the tabbing order.
```
Comment (origami-z) [inline site/docs/components/switch/accessibility.mdx:25]: not sure why here is "within the tabbing order", later is "in the tab order|

**Review by lilyvc — APPROVED** [review]

File: site/docs/components/switch/accessibility.mdx (hunk 6 above)
Comment (feliciackh) [reply site/docs/components/switch/accessibility.mdx:25]: @sion-docs @mikearildbrown do you have a preference how we address this, or are we ok keeping them as is?

Comment (feliciackh) [inline site/docs/components/switch/accessibility.mdx:25]:
```suggestion
- If focus is on Switch, moves focus to the next focusable element in the tab order.
```

**Review by JoanaMMoreira — DISMISSED** [review]

File: site/docs/components/switch/examples.mdx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+- Only use the `checked` prop for a controlled switch that requires some logic for validating value.
+- We recommend that you always use a text description for a switch.
+
+</LivePreview>
+
+<LivePreview componentName="switch" exampleName="DefaultChecked" displayName="Checked" >
+
+### Checked
+
+When `defaultChecked=”true”`, switch is in a `checked` state and the switch thumb sits to the right side of the track.
```
Comment (origami-z) [inline site/docs/components/switch/examples.mdx:33]:
```suggestion
When `defaultChecked={true}`, switch is in a `checked` state and the switch thumb sits to the right side of the track.
```

**Review by origami-z — APPROVED** [review]

**Pull request merged**